cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/data/SparkDataRow.java

```diff
@@ -100,11 +104,29 @@ public Object getValue(String fieldName) {
             } else {
                 result = doDefaultConversion(sparkVal);
             }
+
+            if (metadata.contains(IS_PRIMARY_ANY_COL)) {
```

Comment:
The only potential downside I see so far with metadata is that a "primary any column" can no longer be retrieved on its own. If an offering goes down the metadata approach for any columns, then we have to make sure this shift in retrieve capabilities is spelled out in our documentation.